R/execution.r

```diff
@@ -25,6 +25,18 @@ plot_builds_upon <- function(prev, current) {
     lcurrent >= lprev && identical(current[[1]][1:lprev], prev[[1]][1:lprev])
 }
 
+ask <- function(prompt = '') {
+    answer <- NA
+    while (is.na(answer)) {
+        answer <- switch(readline(prompt),
```

Comment:
This is terminal interaction, right? I think we should avoid anything that does that until we've implemented the stdin part of the protocol. Otherwise it will lead to bizarre hangs when you try to shut it down.